src/main/java/org/commcare/core/network/ModernHttpRequester.java

```diff
@@ -112,9 +116,43 @@ public void makeRequestAndProcess() {
             default:
                 throw new IllegalArgumentException("Invalid HTTPMethod " + method.toString());
         }
-        return currentCall.execute();
+
+        return executeAndCheckCaptivePortals(currentCall);
+    }
+
+    private Response executeAndCheckCaptivePortals(Call currentCall) throws IOException {
+        try {
+            return currentCall.execute();
+        } catch (SSLHandshakeException | SSLPeerUnverifiedException e) {
+            // This may be a real SSL exception associated with the real endpoint server, or this
+            // might be a property of the local network.
+            if(checkCurrentNetworkAsCaptivePortal()) {
+                throw new CaptivePortalRedirectException(e);
+            }
+
+            //Otherwise just rethrow the original exception. Probably a certificate issue
+            throw e;
+        }
     }
 
+    private boolean checkCurrentNetworkAsCaptivePortal() {
+        String captivePortalURL = "http://www.commcarehq.org/serverup.txt";
+        CommCareNetworkService commCareNetworkService =
+                CommCareNetworkServiceGenerator.createNoAuthCommCareNetworkService();
+        try {
+            Response<ResponseBody> response =
+                    commCareNetworkService.makeGetRequest(captivePortalURL, new HashMap<>(), new HashMap<>()).execute();
+            Logger.log(LogTypes.TYPE_WARNING_NETWORK, "Detecting captive portal, Response Code: "
```

Comment:
I think we should remove the log from here and let the catchers of the `CaptivePortalRedirectException` log it in individual processes.